crates/pessimistic-proof/src/local_state.rs

```diff
@@ -99,6 +99,22 @@ impl LocalNetworkState {
             {
                 return Err(ProofError::InvalidImportedBridgeExitRoot);
             }
+
+            match &imported_bridge_exit.inclusion_proof_rer {
+                Some((_proof, rer)) => {
+                    if *rer != multi_batch_header.imported_rollup_exit_root {
+                        return Err(ProofError::InvalidImportedBridgeExitRoot);
+                    }
+                }
+                None => {
+                    if imported_bridge_exit.imported_local_exit_root
+                        != multi_batch_header.imported_mainnet_exit_root
+                    {
+                        return Err(ProofError::InvalidImportedBridgeExitRoot);
+                    }
+                }
```

Comment:
Not sure how important it is, but we could also check here that `imported_bridge_exit.global_index.mainnet_flag` is `false` when `Some` and `true` when `None`.

Reply:
I added the check